src/osqp/tests/non_convex_test.py

```diff
@@ -1,52 +1,60 @@
 from types import SimpleNamespace
 import osqp
-from osqp import constant, default_algebra
-from osqp.tests.utils import SOLVER_TYPES
 import numpy as np
 from scipy import sparse
 
 import pytest
 import numpy.testing as nptest
 
 
-@pytest.fixture(params=SOLVER_TYPES)
-def self(request):
-    self = SimpleNamespace()
-    self.P = sparse.triu([[2.0, 5.0], [5.0, 1.0]], format='csc')
-    self.q = np.array([3, 4])
-    self.A = sparse.csc_matrix([[-1.0, 0.0], [0.0, -1.0], [-1.0, 3.0], [2.0, 5.0], [3.0, 4]])
-    self.u = np.array([0.0, 0.0, -15, 100, 80])
-    self.l = -np.inf * np.ones(len(self.u))
-    self.model = osqp.OSQP()
-    self.model.solver_type = request.param
-    return self
-
-
-@pytest.mark.skipif(default_algebra() != 'builtin', reason='Only applicable for builtin algebra')
-def test_non_convex_small_sigma(self):
-    opts = {'verbose': False, 'sigma': 1e-6}
-    try:
-        # Setup should fail due to (P + sigma I) having a negative
-        # eigenvalue
-        test_setup = 1
-        self.model.setup(P=self.P, q=self.q, A=self.A, l=self.l, u=self.u, **opts)
-    except ValueError:
-        test_setup = 0
-
-    assert test_setup == 0
+@pytest.fixture
+def self(algebra, solver_type, atol, rtol, decimal_tol):
+    ns = SimpleNamespace()
+    ns.P = sparse.triu([[2.0, 5.0], [5.0, 1.0]], format='csc')
+    ns.q = np.array([3, 4])
+    ns.A = sparse.csc_matrix([[-1.0, 0.0], [0.0, -1.0], [-1.0, 3.0], [2.0, 5.0], [3.0, 4]])
+    ns.u = np.array([0.0, 0.0, -15, 100, 80])
+    ns.l = -np.inf * np.ones(len(ns.u))
+    ns.model = osqp.OSQP(algebra=algebra)
+    return ns
+
+
+def test_non_convex_small_sigma(self, solver_type):
+    if solver_type == 'direct':
+        with pytest.raises(ValueError):
+            self.model.setup(
+                P=self.P,
+                q=self.q,
+                A=self.A,
+                l=self.l,
+                u=self.u,
+                solver_type=solver_type,
+                sigma=1e-6,
+            )
+    else:
+        self.model.setup(
+            P=self.P,
+            q=self.q,
+            A=self.A,
+            l=self.l,
+            u=self.u,
+            solver_type=solver_type,
+            sigma=1e-6,
+        )
+        res = self.model.solve()
+
+        if self.model.algebra == 'mkl':  # pytest-todo
```

Comment:
Instead of the algebra check like this, I think you can just test for either status value on every algebra (they are both equally valid results for a solution to a non-convex problem on an indirect solver).

Reply:
done